### ISSUE TITLE
Error while testing a new Model adaptor

### ISSUE THREAD

**Issue by nikhiljohn10** [balderdashy/sails#2988]
Error is as follows.

```
  1) "before all" hook
[Error: timeout of 6000ms exceeded. Ensure the done() callback is being called in this test.]
  2) "after all" hook
[TypeError: Cannot read property 'collections' of undefined]

  0 passing (6s)
  2 failing

  1)  "before all" hook:
     Error: timeout of 6000ms exceeded. Ensure the done() callback is being called in this test.
  

  2)  "after all" hook:
     TypeError: Cannot read property 'collections' of undefined
  ```

This is my adapter code.

```
        syncable: false,
        identity: 'sails-cps',
        defaults: {
            url: 'tcp://cloud-eu-0.clusterpoint.com:9007',
            db: 'Test',
            username: '******',
            password: '***',
            aid: ***
        },
         registerConnection: function(connection, collections, cb) {
            if (!connection.identity) return cb(new Error('Connection is missing an identity.'));
            if (connections[connection.identity]) return cb(new Error('Connection is already registered.'));
			var config = this.defaults; 
            connections[connection.identity] = new cps.Connection(
                config.url,
                config.db,
                config.username,
                config.password,
                'document',
                'document/id', {
                    account: config.aid
                }
            );
        },
        create: function(connectionName, collection, data, cb) {
            var connectionObject = connections[connectionName];
            connectionObject.sendRequest(new cps.InsertRequest(data), function(err, resp) {
                if (err) return cb(err);
                cb(null, resp);
            });
        }

```

**Comment by nikhiljohn10** [comment]
The adapter didn't worked out. So i made a service to handle the same task.

https://github.com/nikhiljohn10/sails-cpsdb-service

**Comment by tjwebb** [comment]
1. Can you post the rest of your adapter code?
2. It'd be awesome to convert that to a bona fide sails adapter once we sort out this issue

**Comment by nikhiljohn10** [comment]
@tjwebb This API is not as flexible as local db. The service i create covers like 95% of that API's functions. what is advantage of a adapter for this API over the service?

**Comment by nikhiljohn10** [comment]
This is the api is used.

https://github.com/clusterpoint/nodejs-client-api

**Comment by tjwebb** [comment]
You don't need to implement every adapter function. I'm not familiar with clusterpoint; what is the goal here?

**Comment by sailsbot** [comment]
Thanks for posting, @nikhiljohn10.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
